mbrl/models/model.py

```diff
@@ -155,15 +172,34 @@ def update(
 
         Returns:
              (float): the numeric value of the computed loss.
-
+             (dict): any additional metadata dictionary computed by :meth:`loss`.
         """
         optimizer = cast(torch.optim.Optimizer, optimizer)
         self.train()
         optimizer.zero_grad()
-        loss = self.loss(model_in, target)
-        loss.backward()
-        optimizer.step()
-        return loss.item()
+        loss_and_maybe_meta = self.loss(model_in, target)
+        if isinstance(loss_and_maybe_meta, tuple):
+            # TODO - v0.2.0 remove this back-compatibility logic
+            loss = cast(torch.Tensor, loss_and_maybe_meta[0])
+            meta = cast(Optional[Dict[str, Any]], loss_and_maybe_meta[1])
+            loss.backward()
+
+            if meta is not None:
+                with torch.no_grad():
+                    grad_norm: torch.Tensor = 0  # type: ignore
```

Comment:
```suggestion
                    grad_norm: float = 0
```
Both here and in line 193 a float is assigned to `grad_norm` and never a tensor, so the type should be `float`.

Reply:
Fixed.